app/javascript/lib/Entities/Board/Province.js

```diff
@@ -0,0 +1,65 @@
+import Entity from '../Entity';
+
+export default class Province extends Entity {
+  constructor(isLand, representation) {
+    super(representation);
+
+    this.isLand = isLand;
+
+    this.ownership = null;
+    this.hasArmsFactory = false;
+    this.hasNavalFactory = false;
+
+    this.friendlyUnits = new Map();
+    this.hostileUnits = new Map();
+  }
+
+  get isOcean() {
+    return this.isLand === false;
+  }
+  set isOcean(bool) {
+    this.isLand = bool === false;
+  }
+
+  get hasFactory() {
+    return this.hasArmsFactory || this.hasNavalFactory;
+  }
+  get isDevoidOfFactory() {
+    return this.hasFactory === false;
+  }
+
+  get isOccupiedByHostiles() {
+    return this.hostileUnits.size > 0;
+  }
+  get isNotOccupiedByHostiles() {
+    return this.isOccupiedByHostiles === false;
+  }
+}
+
+export const translateProvinceModel = (representation, allProvinces, allUnits, gameBoard) => {
+  // temporary until migration is complete
+  const oldProvinceModel = allProvinces.get(representation);
+  const geographicModel = gameBoard.graph.get(representation);
+
+  const province = new Province(
+    geographicModel.isOcean === false || geographicModel.isOcean === undefined,
```

Comment:
Super small point here but couldn't we do `new Province(!geographicModel.isOcean, representation)`?

Reply:
Yeah, personally I find the `!` kinda small and easy to miss, and therefore prefer `== false`, except that's not necessarily true here since `js` shenanigans

Will refactor to `! ... .isOcean` since it's clear what it's doing anyway

Reply:
[2a152d3](https://github.com/Thrillberg/imperial/pull/711/commits/2a152d342ba4c7edb3f493060703c859b8ceacd5)